### PR TITLE
Add org.kde.peruse

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/graphics/peruse/-/merge_requests/37
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://invent.kde.org/graphics/peruse/-/merge_requests/37

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.kde.peruse

**Comment by hfiguiere** [comment]
cc @flathub/kde

**Comment by tsdgeos** [comment]
You want okular QML compiled for support PDF files at least

**Comment by Mailaender** [comment]
Can't build right now, as https://release-monitoring.org/ is down.

**Comment by hfiguiere** [comment]
bot, build org.kde.peruse

File: org.kde.peruse.json [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+                                "url-template": "https://github.com/gnustep/tools-make/archive/make-$version.tar.gz"
+                            }
+                        }
+                    ]
+                },
+                {
+                    "name": "gnustep",
+                    "buildsystem": "simple",
+                    "build-commands": [
+                        "./configure --prefix=/app",
```
Comment (hfiguiere) [inline org.kde.peruse.json:85]: why "simple" and not autotools?

File: org.kde.peruse.json [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+                        "make -j $FLATPAK_BUILDER_N_JOBS install",
+                        "ls /app/include"
```
Comment (hfiguiere) [inline org.kde.peruse.json:87]: why this ls?

File: org.kde.peruse.json [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+                            }
+                        }
+                    ]
+                },
+                {
+                    "name": "unarchiver",
+                    "buildsystem": "simple",
+                    "build-commands": [
+                        "sed -s -i 's#/usr/include/GNUstep#/app/include#g' */Makefile.linux && cd XADMaster && make -j $FLATPAK_BUILDER_N_JOBS -f Makefile.linux",
+                        "install -m755 XADMaster/unar XADMaster/lsar /app/bin/"
```
Comment (hfiguiere) [inline org.kde.peruse.json:108]:
```suggestion
                        "install -Dm755 XADMaster/unar XADMaster/lsar -t /app/bin/"
```

File: org.kde.peruse.json [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+                            "sha256": "ac4cedf9b49375c0f805bda010fa03629d05ab283a0a8e35bf2dd01ae477e0a1",
+                            "strip-components": 0
+                        }
+                    ]
+                },
+                {
+                    "name": "discount",
+                    "buildsystem": "simple",
+                    "build-commands": [
+                        "./configure.sh --prefix=/app --enable-all-features --with-fenced-code --shared",
```
Comment (hfiguiere) [inline org.kde.peruse.json:123]: why not autotools?

You can do the sed with a patch as well. it's safer.

File: org.kde.peruse.json [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+        "--share=network",
+        "--device=dri",
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--talk-name=org.kde.kcookiejar5",
+        "--system-talk-name=org.freedesktop.UDisks2"
+    ],
+    "modules": [
+        {
+            "name": "peruse",
```
Comment (hfiguiere) [inline org.kde.peruse.json:19]: since this is the main app it should be last in the module list.

File: org.kde.peruse.json (hunk 4 above)
Comment (Mailaender) [reply org.kde.peruse.json:123]: The sed can't be done before configure and configure is not configure but ./configure.sh

**Comment by Mailaender** [comment]
bot, build org.kde.peruse

**Comment by Justinzobel** [comment]
I tried this but it closed when loading a cbz comic book file:
```
qml: Mimetype is application/vnd.comicbook+zip
QFSFileEngine::open: No file name specified
file:///usr/lib/qml/org/kde/kirigami.2/templates/OverlaySheet.qml:246:38: QML FocusScope: Binding loop detected for property "contentItemMaximumWidth"
qrc:/qml/Book.qml:377:13: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
```

**Comment by hfiguiere** [comment]
did it crash?

**Comment by Justinzobel** [comment]
Yeah it looks that way, it shows in coredumpctl. I'll try get a backtrace.

**Comment by Justinzobel** [comment]
I tried but it's missing symbols and I guess the test builds don't generate the Debug symbols package?

**Comment by hfiguiere** [comment]
they do. but you need install the `.Debug` package for that.

**Comment by Justinzobel** [comment]
> they do. but you need install the `.Debug` package for that.

Yeah it's just not available when built as a test package I think. I tried `flatpak install --user https://dl.flathub.org/build-repo/40398/org.kde.peruse.Debug.flatpakref` but it comes back with a 404.

**Comment by hfiguiere** [comment]
that's not how it works. 


`flatpak install org.kde.peruse.Debug` should find it.

**Comment by Justinzobel** [comment]
> flatpak install org.kde.peruse.Debug

I tried but no dice as there's no remote yet until it's merged?
```
flatpak --user install org.kde.peruse.Debug
Looking for matches…
error: No remote refs found for ‘org.kde.peruse.Debug’
```

**Comment by hfiguiere** [comment]
Try this
```
flatpak install peruse-origin org.kde.peruse.Debug
```

(assuming the remote has the default name, which is key here)

**Comment by Justinzobel** [comment]
Unfortunately, the build has now been deleted so I can't download it and test again.

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
bot, build org.kde.peruse

**Comment by Justinzobel** [comment]
Ping @Mailaender

**Comment by Mailaender** [comment]
What do you expect from me? I am done here.

**Comment by bbhtt** [comment]
Add the following cleanups:

```
- "*.a"
- ".la"
- /etc
- /include
- /lib/cmake
- /lib/pkgconfig
- /mkspecs
- /share/doc
- /share/man
- /share/pkgconfig
- /share/qlogging-categories5
```

Okular installs multiple desktop file, one for each file format is supports, those probably should also be cleaned up eg. `org.kde.mobile.okular_pdf.desktop`.

File: org.kde.peruse.json [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+                                "type": "anitya",
+                                "project-id": 12139,
+                                "stable-only": true,
+                                "url-template": "https://github.com/Orc/discount/archive/v$version.tar.gz"
+                            }
+                        }
+                    ]
+                },
+                {
+                    "name": "boost",
```
Comment (bbhtt) [inline org.kde.peruse.json:120]: Seems boost is only needed by poppler and it is a only a make depends of poppler. Can you try cleaning it up completely and check if everything works?

**Comment by Mailaender** [comment]
bot, build org.kde.peruse

**Comment by bbhtt** [comment]
Seems to work fine

> I tried this but it closed when loading a cbz comic book file

cbz files seem to work fine for me, tested with [this](https://github.com/filestar/File-Type-Sample-Repository/blob/master/cbz/sample.cbz?raw=true)

**Pull request closed without merging**